sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -472,16 +475,72 @@ public <T> CosmosPagedFlux<T> queryItems(String query, Class<T> classType) {
     public Mono<Void> openConnectionsAndInitCaches() {
 
         if(isInitialized.compareAndSet(false, true)) {
-            return withContext(context -> openConnectionsAndInitCachesInternal()
+
+            CosmosContainerIdentity cosmosContainerIdentity = new CosmosContainerIdentity(this.database.getId(), this.id);
+            ProactiveContainerInitConfig proactiveContainerInitConfig = new ProactiveContainerInitConfigBuilder(Arrays.asList(cosmosContainerIdentity))
+                    .setProactiveConnectionRegions(1)
+                    .build();
+
+            return withContext(context -> openConnectionsAndInitCachesInternal(proactiveContainerInitConfig)
                                             .flatMap(openResult -> {
                                                 logger.info("OpenConnectionsAndInitCaches: {}", openResult);
                                                 return Mono.empty();
                                             }));
         } else {
-            logger.warn(
-                    String.format(
-                        "OpenConnectionsAndInitCaches is already called once on Container %s, no operation will take place in this call",
-                        this.getId()));
+            logger.warn("OpenConnectionsAndInitCaches is already called once on Container {}, no operation will take place in this call", this.getId());
+            return Mono.empty();
+        }
+    }
+
+    /**
+     *  Best effort to initialize the container by warming up the caches and connections to a specified no. of regions from the
+     *  preferred list of regions.
+     *
+     *  Depending on how many partitions the container has, the total time needed will also change. But generally you can use the following formula
+     *  to get an estimated time:
+     *  If it took 200ms to establish a connection, and you have 100 partitions in your container
+     *  then it will take around (100 * 4 / CPUCores) * 200ms to open all connections after get the address list
+     *
+     *  <p>
+     *  <br>NOTE: This API ideally should be called only once during application initialization before any workload.
+     *  <br>In case of any transient error, caller should consume the error and continue the regular workload.
+     *  </p>
+     * <p>
+     * In order to minimize latencies associated with warming up caches and opening connections
+     * the no. of proactive connection regions cannot be more than {@link ProactiveContainerInitConfigBuilder#MAX_NO_OF_PROACTIVE_CONNECTION_REGIONS}.
```

Comment:
Why the limitation here? (sorry I forget the context here if we have discussed about it) -> eventually it means if the client is in remote region, and if they want to open connections to write region, then they have to move write region up to the 2nd places, I feel it could be too strict

Reply:
+1 - I also feel 2 is too restrictive - having a restriction isn't terrible - but maybe set it to 5 for now?